client/lib/i18n-utils/switch-locale.js

```diff
@@ -107,10 +108,62 @@ export default function switchLocale( localeSlug ) {
 			i18n.setLocale( response.body );
 
 			setLocaleInDOM( domLocaleSlug, !! language.rtl );
+
+			applyUserWaitingTranslations( targetLocaleSlug );
 		} );
 	}
 }
 
+function applyUserWaitingTranslations( currentLocaleSlug ) {
+	if ( ! location || ! location.search ) {
+		return;
+	}
+
+	const parsedURL = parseUrl( location.search, true );
+
+	const {
+		'load-user-translations': username,
+		project = 'wpcom',
+		translationSet = 'default',
+		locale = currentLocaleSlug,
+	} = parsedURL.query;
+
+	if ( ! username ) {
+		return;
+	}
+
+	const pathname = [
+		'api',
+		'projects',
+		project,
+		locale,
+		translationSet,
+		'export-translations',
+	].join( '/' );
+
+	const query = {
+		'filters[user_login]': username,
+		'filters[status]': 'waiting',
```

Comment:
I think we'll need to use `current_or_waiting_or_fuzzy_or_untranslated` for now. The main use case is actually for reviewers to be able to disconnect from i18n-deploy, so their translations will be `current` thus not included in this query.